File: src/main/java/net/sharplab/epubtranslator/core/model/EPubChapter.java
package net.sharplab.epubtranslator.core.model;

import java.nio.charset.StandardCharsets;

/**
 * EPubChapter
 */
public class EPubChapter extends EPubContentFile {

    public EPubChapter(String name, byte[] data) {
        super(name, data);
    }

    public String getDataAsString() {
        return new String(getData(), StandardCharsets.UTF_8);
    }
}

File: src/main/java/net/sharplab/epubtranslator/core/model/EPubChapter.kt
package net.sharplab.epubtranslator.core.model

import java.nio.charset.StandardCharsets

/**
 * EPubChapter
 */
class EPubChapter(name: String, data: ByteArray) : EPubContentFile(name, data) {
    val dataAsString: String
        get() = String(data, StandardCharsets.UTF_8)
}
